ui/pages/confirmations/components/confirm/info/approve/edit-spending-cap-modal/edit-spending-cap-modal.tsx

```diff
@@ -0,0 +1,177 @@
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { updateEditableParams } from '../../../../../../../store/actions';
+import { getCustomTxParamsData } from '../../../../../confirm-approve/confirm-approve.util';
+import { useConfirmContext } from '../../../../../context/confirm';
+import { useAssetDetails } from '../../../../../hooks/useAssetDetails';
+import { useApproveTokenSimulation } from '../hooks/use-approve-token-simulation';
+import { useTransactionGasEstimate } from '../hooks/use-transaction-gas-estimate';
+
+export const getAccountBalance = (userBalance: string, decimals: string) =>
+  new BigNumber(userBalance)
+    .dividedBy(new BigNumber(10).pow(Number(decimals)))
+    .toNumber();
+
+export const EditSpendingCapModal = ({
+  isOpenEditSpendingCapModal,
+  setIsOpenEditSpendingCapModal,
+}: {
+  isOpenEditSpendingCapModal: boolean;
+  setIsOpenEditSpendingCapModal: (newValue: boolean) => void;
+}) => {
+  const t = useI18nContext();
+
+  const dispatch = useDispatch();
+
+  const { currentConfirmation: transactionMeta } = useConfirmContext() as {
+    currentConfirmation: TransactionMeta;
+  };
+
+  const { userBalance, tokenSymbol, decimals } = useAssetDetails(
+    transactionMeta.txParams.to,
+    transactionMeta.txParams.from,
+    transactionMeta.txParams.data,
+  );
+
+  const accountBalance = getAccountBalance(userBalance || '0', decimals || '0');
+
+  const { formattedSpendingCap } = useApproveTokenSimulation(
+    transactionMeta,
+    decimals || '0',
+  );
+
+  const [customSpendingCapInputValue, setCustomSpendingCapInputValue] =
+    useState('');
+  const [customSpendingCap, setCustomSpendingCap] = useState('');
+  const setPositiveSpendingCap = (newValue: string) => {
+    const parsedValue = parseInt(newValue, 10);
+    // coerce negative numbers to zero
+    setCustomSpendingCap(parsedValue < 0 ? '0' : newValue);
+  };
+
+  const customTxParamsData = useMemo(() => {
+    return getCustomTxParamsData(transactionMeta?.txParams?.data, {
+      customPermissionAmount: customSpendingCap || '0',
+      decimals: decimals || '0',
+    });
+  }, [customSpendingCap, transactionMeta?.txParams?.data, decimals]);
+
+  const { estimatedGasLimit } = useTransactionGasEstimate(
+    transactionMeta,
+    customTxParamsData,
+    customSpendingCap,
+  );
+
+  useEffect(() => {
+    if (customSpendingCap && estimatedGasLimit) {
+      dispatch(
+        updateEditableParams(transactionMeta.id, {
+          data: customTxParamsData,
+          gas: hexToDecimal(estimatedGasLimit as string),
+        }),
+      );
+    }
+  }, [
+    customSpendingCap,
+    estimatedGasLimit,
+    customTxParamsData,
+    transactionMeta,
+    dispatch,
+  ]);
+
+  const handleCancel = () => {
+    setIsOpenEditSpendingCapModal(false);
+    setCustomSpendingCapInputValue('');
+  };
+
+  function handleSubmit() {
+    setIsOpenEditSpendingCapModal(false);
+    setCustomSpendingCapInputValue('');
```

Comment:
Does `handleSubmit` automatically close the modal as part of the modal component itself?

In that case, can we skip clearing the state and just validate and `dispatch`?

Reply:
I believe we still want to clear the input value state, to default to show the placeholder next time the modal is opened. Keep in mind this component is not unmounted, so we need to manage the state hook lifecycle.